librarian_server/api/admin.py

```diff
@@ -96,3 +100,47 @@ def add_file(
     session.commit()
 
     return AdminCreateFileResponse(success=True, file_exists=True)
+
+
+@router.post("/verify_file", response_model=FileVerificationResponse)
+def verify_file(
+    request: AdminVerifyFileRequest,
+    session: Session = Depends(yield_session),
+) -> Dict[str, List[Dict[str, str]]]:
+    """
+    Verifies the properties of an existing file in the database and returns newly computed checksums and sizes for all of the instances.
+    If the requested store or file does not exist, or if no instances are found, it returns a 400 Bad Request response.
+    """
+
+    store = (
+        session.query(StoreMetadata).filter_by(name=request.store_name).one_or_none()
+    )
+    if store is None:
+        raise HTTPException(status_code=400, detail="Store not found.")
+
+    file = session.query(File).filter_by(name=request.name).one_or_none()
+    if file is None:
+        raise HTTPException(status_code=400, detail="File not found.")
+
+    instances = (
+        session.query(Instance)
+        .join(File, File.name == Instance.file_name)
+        .filter(File.name == file.name, Instance.store_id == store.id)
+        .all()
+    )
+    if not instances:
+        raise HTTPException(status_code=400, detail="File instances not found.")
+
+    checksums_and_sizes = []
+    for instance in instances:
+        path = instance.path
+        checksum = get_md5_from_path(path)
+        size = get_size_from_path(path)
+        checksums_and_sizes.append(
+            {
+                "store_id": str(instance.store_id),
+                "checksum": checksum,
+                "size": str(size),
+            }
+        )
+    return {"verified": True, "checksums_and_sizes": checksums_and_sizes}
```

Comment:
We never return unvalidated JSON. Please write a pydantic model for your return.